### PR TITLE
Move layerDependencies to formatter

### PR DESCRIPTION
**Pull request by elpaso**
... from the widget

### CODE REVIEW

**Comment by elpaso** [comment]
@m-kuhn is that what you had in mind (still to be checked and polished!)

File: src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include "qgseditorwidgetwrapper.h"
 #include "qgis_sip.h"
 #include "qgis_gui.h"
+#include "qgsvectorlayerref.h"
```
Comment (m-kuhn) [inline src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.h]: That's probably a leftover?

**Review by m-kuhn — COMMENTED** [review]
Yes, that looks exactly like what I had in mind. Thanks for the quick reaction!

File: src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.h (hunk 1 above)
Comment (elpaso) [reply src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.h]: yes, thanks, I said it needed polishing (ans tests!)

Comment (m-kuhn) [reply src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.h]: sure, just bumped into it.
I hope the tests can be easily adapted.

Comment (elpaso) [reply src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.h]: I'm not sure, QgsVectorLayerRef is not exposed to Python so I need a C++ test and there aren't any for the formatters.

Comment (m-kuhn) [reply src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.h]: just copying / renaming the ones you did for widgetwrappers won't work?

Comment (elpaso) [reply src/gui/editorwidgets/qgsrelationreferencewidgetwrapper.h]: I'm creating a new `testqgsrelationreferencefieldformatter.cpp` and moving the old test into it.

**Pull request merged**